broker/server.js

```diff
@@ -104,102 +104,127 @@ app.post("/Encounter", (req, res) => {
-      `?subject=${req.query.patient}&status=planned`,
-      encounter_status,
-      res
-    );
-    updateAppointmentStatus(
-      `?actor=${req.query.patient}&status=booked`,
-      appointment_status,
-      res
-    );
-    res.send("");
+    encounter_url = `?subject=${req.query.patient}&status=planned`;
+    appt_url = `?actor=${req.query.patient}&status=booked`;
   } else if (req.query.hasOwnProperty("appointment")) {
-    updateEncounterStatus(
-      `?appointment=${req.query.appointment}`,
-      encounter_status,
-      res
-    );
-    updateAppointmentStatus(
-      `/${req.query.appointment}`,
-      appointment_status,
-      res
-    );
-    res.send("");
+    encounter_url = `?appointment=${req.query.appointment}`;
+    appt_url = `/${req.query.appointment}`;
   } else {
-    res.send("error");
+    response.error = "patient or appointment parameter missing";
+    res.status(400).json(response);
+    return;
   }
+
+  Promise.all([
+    updateEncounterStatus(encounter_url, encounter_status),
+    updateAppointmentStatus(appt_url, appt_status),
+  ])
+    .then((results) => {
+      for (result of results) {
+        response[result.data.resourceType] = result.data;
+      }
+      res.json(response);
+    })
+    .catch((e) => {
+      response.error = e.response ? e.response.data : e.message;
+      res.status(400).json(response);
+    });
 });
 
 // Given url of Encounter and new status, update encounter resource
 // URL is either a specific id '/id'
 // or query parameters '?param=value'
-function updateEncounterStatus(url, status, res) {
-  axios
-    .get(`${base}/Encounter${url}`)
-    .then((response) => {
-      try {
-        let encounter;
-        if (url.startsWith("/")) {
-          encounter = response.data;
-        } else {
-          let bundle = response.data;
-          // update status
-          encounter = bundle.entry[0].resource;
-        }
-        encounter.status = status;
-        // update status history
-
-        // update the database with new encounter
-        axios
-          .put(`${base}/Encounter/${encounter.id}`, encounter, headers)
-          .catch((error) => handleError(res, error));
-      } catch (e) {
-        console.log(e);
+function updateEncounterStatus(url, status) {
+  return axios.get(`${base}/Encounter${url}`).then((response) => {
+    let encounter;
+    let resourceType = response.data.resourceType;
+
+    if (resourceType === "Encounter") {
+      encounter = response.data;
+    } else {
+      let bundle = response.data;
+
+      if (!bundle.hasOwnProperty("entry")) {
+        return "Encounter does not exist";
       }
-    })
-    .catch((error) => handleError(res, error))
-    .then(function () {
-      // always executed
-    });
+      encounter = bundle.entry[0].resource;
+    }
+    encounter.status = status;
+
+    // update the database with new encounter
+    return axios
+      .put(`${base}/Encounter/${encounter.id}`, encounter, headers)
```

Comment:
If you make this a PATCH instead of a PUT, you don't need to pass the entire resource back to the server; you can just pass back the elements you want to change. Then you wouldn't need to make the GET request first thing in this function. Then again, doing the GET first allows you to do some validation. Either way, just wanted to let you know that PATCH is an option if it can simplify things any.